### ISSUE TITLE
support kl_div function with bfloat16

### ISSUE THREAD

**Issue by rosrad** [pytorch/pytorch#77375]
### 🚀 The feature, motivation and pitch

We are trying to switch our model training from the fp16 to bfloat16 precision. This change can be good for the CE loss functions, however it failed on the kl_div function due to non implementation.

here is the error log:
`"kl_div_backward_cuda" not implemented for 'BFloat16'`

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by ngimel** [comment]
cc @ptrblck, can someone send a PR implementing this please?

**Comment by ptrblck** [comment]
@Aidyn-A could you take a look at this, please?

**Comment by Aidyn-A** [comment]
Sure, I will take care of this.

**Comment by boren-ms** [comment]
Thanks a lot for the timely support.
